src/components/login/AuthDetails.tsx

```diff
@@ -0,0 +1,28 @@
+import { onAuthStateChanged, signOut } from "firebase/auth";
+import { useEffect, useState } from "react";
+import { auth } from '../../firebase-config';
+
+const AuthDetails = () => {
+    const [authUser, setAuthUser] = useState(null);
+
+    useEffect(() => {
+        const listen = onAuthStateChanged(auth, (user) => {
+            if (user) {
+                setAuthUser
+            } else {
+                setAuthUser(null);
+            }
+        })
+        return () => listen();
```

Comment:
Also, if listen is changed to removeAuthListener, make sure to update this line too. Like this:
`return () => removeAuthListener();`

Reply:
done